### PR TITLE
fix(query): Revise list of unsafe sysctls in cluster_allows_unsafe_sysctls k8s rule

### PR DESCRIPTION
**Pull request by Churro**
**Problem**

* Many containers run as root so that they can bind on ports < 1024. As this poses a certain security risk, Kubernetes v1.22 introduced the possibility to specify `net.ipv4.ip_unprivileged_port_start` in the sysctls of the securityContext of a pod. This enables also unprivileged users to bind on privileged ports. The K8s documentation therefore includes this sysctl in the safe set (see [here](https://kubernetes.io/docs/tasks/administer-cluster/sysctl-cluster/#enabling-unsafe-sysctls)). The current rule is not aware of that.
* Since Kubernetes v1.23, sysctl names may alternatively be spelled with a / as separator instead of a dot.   
This scheme is currently not supported.
* The current rule is not immediately actionable, as it doesn't point to the specific line with the problematic sysctl name

**Proposed Changes**

* Remediate false positive in case `net.ipv4.ip_unprivileged_port_start` is specified
* Support sysctl names with /
* Show targeted findings for each statement violating the rule
* Extend rule to also cover Deployment, StatefulSet, etc. resource kinds

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

**Review by rafaela-soares — CHANGES_REQUESTED** [review]
Another great observation and refactor, @Churro 🚀 Thank you!

I only have a suggestion. If you agree, can you also add a negative sample related to the `positive2.yaml`, please? (It is not extremely required, but the more samples, the better)

File: assets/queries/k8s/cluster_allows_unsafe_sysctls/query.rego [hunk 1]
```diff
@@ -11,33 +13,37 @@ CxPolicy[result] {
...
-	metadata := document.metadata
-
 	result := {
 		"documentId": document.id,
-		"searchKey": sprintf("metadata.name=%s.spec.securityContext.sysctls", [metadata.name]),
+		"searchKey": sprintf("metadata.name={{%s}}.%s.securityContext.sysctls.name={{%s}}", [metadata.name, specInfo.path, sysctl]),
 		"issueType": "IncorrectValue",
-		"keyExpectedValue": sprintf("metadata.name=%s.spec.securityContext.sysctls does not have an unsafe sysctl", [metadata.name]),
-		"keyActualValue": sprintf("metadata.name=%s.spec.securityContext.sysctls has an unsafe sysctl", [metadata.name]),
+		"keyExpectedValue": sprintf("metadata.name={{%s}}.%s.securityContext.sysctls.name={{%s}} is not used", [metadata.name, specInfo.path, sysctl]),
```
[the remark this answers is not in the document]
Comment (Churro) [reply assets/queries/k8s/cluster_allows_unsafe_sysctls/query.rego:35]: I appreciate your suggestion but I think there are arguments against rephrasing it like this:
E.g., with unsafe sysctls we would see alerts in the form: `keyExpectedValue: ...securityContext.sysctls.name={{net.core.somaxconn}} is not unsafe`. The issue here is that `{{net.core.somaxconn}}` cannot be *not* unsafe.

In other words, remediation is not possible by changing the value set for this sysctl but only by not using the unsafe sysctl at all. Therefore, I think that "<unsafe sysctl> is not used" would be a better suited statement than "<unsafe sysctl> is not unsafe".

**Comment by Churro** [comment]
Thank you for the review, @rafaela-soares! As request, I added a negative test sample.

File: assets/queries/k8s/cluster_allows_unsafe_sysctls/query.rego (hunk 1 above)
[the remark this answers is not in the document]
Comment (rafaela-soares) [reply assets/queries/k8s/cluster_allows_unsafe_sysctls/query.rego:35]: I understood! Makes sense! Thank you so much for the explanation and the negative test sample addition, @Churro!

**Review by rafaela-soares — APPROVED** [review]
LGTM

**Pull request merged**